nifi-commons/nifi-record/src/test/java/org/apache/nifi/serialization/record/TestDataTypeUtils.java

```diff
@@ -442,6 +442,14 @@ private void testChooseDataType(List<DataType> dataTypes, Object value, DataType
         assertEquals(expected, actual);
     }
 
+    @Test
+    public void testInferTypeWithMapNonStringKeys() {
+        Map<Integer, String> map = new HashMap<>();
+        map.put(1, "Hello");
+        map.put(2, "World");
+        DataTypeUtils.inferDataType(map, RecordFieldType.MAP.getMapDataType(RecordFieldType.STRING.getDataType()));
```

Comment:
An `expected` could be useful as this doesn't test the string keys only the lack of exception. (Later a bug could make it return an empty map or null for example that this test would not catch.)
```java
        RecordDataType expected = (RecordDataType)RecordFieldType.RECORD.getRecordDataType(new SimpleRecordSchema(Arrays.asList(
            new RecordField("1", RecordFieldType.STRING.getDataType()),
            new RecordField("2", RecordFieldType.STRING.getDataType())
        )));

...

        DataType actual = DataTypeUtils.inferDataType(map, null);

        assertEquals(expected, actual);
```

Also a minor suggestion: could simply be
```java
        DataTypeUtils.inferDataType(map, null);
```
(The second parameter is just a default type and the fact that it's correctly constructed kind of suggests that it's relevant for the test.)

Reply:
Good suggestion, thanks! Done.